app/helpers/search_helper.rb

```diff
@@ -1,9 +1,15 @@
+# frozen_string_literal: true
+
 module SearchHelper
-  def method_anchor(method)
-    # See https://github.com/ruby/rdoc/blob/c64210219ec6c0f447b4c66c2c3556cfe462993f/lib/rdoc/method_attr.rb#L294
-    method_name = CGI.escape(method.name.gsub("-", "-2D")).tr("%", "-").sub(/^-/, "")
+  def escape_method_name(method_name)
+    (@escaped_method_names ||= {})[method_name] ||= begin
```

Comment:
It's not shared between threads is it?

Reply:
I know you already merged this, but no. Each Rails request instantiates a new instance of the view / controller, which the helper methods get mixed into.

I learned a lot from your video about thread safety yesterday. I would have definitely used Concurrent Ruby if it were shared between threads. That’s another optimization that could be made though — because there’s really no point in figuring out the escaped form of the same method name for every Ruby version and class, every single request.

If you’re fine with retaining a little more memory throughout the life of the app, I could make a PR for that change as well!